src/cc/flwr/include/typing.h

```diff
@@ -0,0 +1,316 @@
+		{
+			return bytes;
+		}
+		std::optional<float> getFloat()
+		{
+			return f;
+		}
+		std::optional<int> getInt()
+		{
+			return i;
+		}
+		std::optional<std::string> getString()
+		{
+			return string;
+		}
+
+		// Setters
+		void setBool(bool b)
+		{
+			this->b = b;
+		}
+		void setBytes(std::string bytes)
+		{
+			this->bytes = bytes;
+		}
+		void setFloat(float f)
+		{
+			this->f = f;
+		}
+		void setInt(int i)
+		{
+			this->i = i;
+		}
+		void setString(std::string string)
+		{
+			this->string = string;
+		}
+
+	private:
+		std::optional<bool> b = std::nullopt;
+		std::optional<std::string> bytes = std::nullopt;
+		std::optional<float> f = std::nullopt;
+		std::optional<int> i = std::nullopt;
+		std::optional<std::string> string = std::nullopt;
+	};
+
+	typedef std::map<std::string, flwr::Scalar> Metrics;
+
+	/**
+	* Model parameters
+	*/
+	class Parameters {
+	public:
+		Parameters() {};
+		Parameters(std::list<std::string> tensors, std::string tensor_type)
+			: tensors(tensors), tensor_type(tensor_type) {};
+
+		// Getters
+		std::list<std::string> getTensors()
+		{
+			return tensors;
+		}
+		std::string getTensor_type()
+		{
+			return tensor_type;
+		}
+
+		// Setters
+		void setTensors(std::list<std::string> tensors)
+		{
+			this->tensors = tensors;
+		}
+		void setTensor_type(std::string tensor_type)
+		{
+			this->tensor_type = tensor_type;
+		}
+
+	private:
+		std::list<std::string> tensors;
+		std::string tensor_type;
+	};
+
+	/**
+	* Response when asked to return parameters
+	*/
+	class ParametersRes
+	{
+	public:
+		ParametersRes(Parameters parameters)
+			: parameters(parameters) {};
+
+		Parameters getParameters()
+		{
+			return parameters;
+		}
+		void setParameters(Parameters p)
+		{
+			parameters = p;
+		}
+
+	private:
+		Parameters parameters;
+	};
+
+	/**
+	* Fit instructions for a client
+	*/
+	class FitIns
+	{
+	public:
+		FitIns(Parameters parameters, std::map<std::string, flwr::Scalar> config)
+			: parameters(parameters), config(config) {};
+
+		// Getters
+		Parameters getParameters()
+		{
+			return parameters;
+		}
+		std::map<std::string, Scalar> getConfig()
+		{
+			return config;
+		}
+
+		// Setters
+		void setParameters(Parameters p)
+		{
+			parameters = p;
+		}
+		void setConfig(std::map<std::string, Scalar> config)
+		{
+			this->config = config;
+		}
+
+	private:
+		Parameters parameters;
+		std::map<std::string, Scalar> config;
+	};
+
+	/**
+	* Fit response from a client
+	*/
+	class FitRes
+	{
+	public:
+		FitRes() {};
+		FitRes(Parameters parameters, int num_examples, int num_examples_ceil, float fit_duration, Metrics metrics)
+			: parameters(parameters), num_examples(num_examples),
+			fit_duration(fit_duration), metrics(metrics) {};
+
+		// Getters
+		Parameters getParameters()
+		{
+			return parameters;
+		}
+		int getNum_example()
+		{
+			return num_examples;
+		}
+		/*std::optional<int> getNum_examples_ceil()
+		{
+			return num_examples_ceil;
+		}*/
```

Comment:
Do we need this here? If its not used I would say lets remove it. Its deprecated in the Python code anyway.